lib/server.js

```diff
@@ -1,21 +1,29 @@
 "use strict";
 
-//Modules required
-var Hapi = require("hapi");
+var Hapi              = require("hapi");
+var Bluebird          = require("bluebird");
+var MongoMapper       = require("genesis").MongoMapper;
+var DatabaseConnector = require("./services/DatabaseConnector");
 
-//Setting the port for the application
-var port = process.env.PORT || 8080;
+var databaseConnector = DatabaseConnector.connect(process.env.DB_URL, process.env.DB_USER, process.env.DB_PASSWORD);
+var mapper            = new MongoMapper(databaseConnector);
+
+Bluebird.promisifyAll(Hapi);
+
+var port   = process.env.PORT || 8080;
+var server = module.exports = new Hapi.Server();
 
-//Server initialization
-var server = new Hapi.Server();
 server.connection({
 	host : process.env.HOST,
 	port : port
 });
 
-server.register ([
+server.registerAsync ([
```

Comment:
By using the `Async` version of this function, you are transitioning from a callback-based flow to a promise-based flow. That's good, but the rest of this block has not been converted to use promises instead of callbacks.